public/tests.py

```diff
@@ -1,3 +1,86 @@
-from django.test import TestCase
+import requests
+from django.core.exceptions import ValidationError
+from django.core.validators import URLValidator
+from django.test import TestCase, SimpleTestCase
 
-# Create your tests here.
+from library_website.settings import IDRESOLVE_URL
+from public.utils import doi_lookup, doi_lookup_base_url, mk_url
+
+example_doi1 = "10.1007/s11050-007-9022-y"
+example_doi2 = "10.1017/S0960129518000324"
+bad_url = "https://not-an-actual-api-I-just-made-it-up.com?"
+bad_doi = "NOT_A_DOI"
+
+
+class IdresolveTest(SimpleTestCase):
+
+    fixtures = ['test.json']
+
+    def test_idresolve_is_up(self):
+        """
+        check that idresolve returns a response when given a valid DOI and
+        base URL
+
+        """
+        assert doi_lookup(example_doi1) is not None
+        assert doi_lookup(example_doi2) is not None
+
+    def test_good_url_no_exception(self):
+        """
+        ensure no exception gets thrown when trying to connect to
+        idresolve at the correct URL
+
+        """
+        try:
+            doi_lookup(example_doi1)
+            doi_lookup(example_doi2)
+        except requests.ConnectionError:
+            self.fail("Idresolve service is not up.")
+
+    def test_bad_url_no_exception(self):
+        """
+        ensure no exception gets thrown when trying to connect to
+        idresolve at the wrong URL
+        """
+        try:
+            doi_lookup_base_url(example_doi1, bad_url)
+            doi_lookup_base_url(example_doi2, bad_url)
+        except requests.ConnectionError as exception:
+            self.fail("public.utils doi_lookup raised %s"
+                      % str(exception)
+                      )
```

Comment:
I like the way you handled this. This is a very good test and a nice way of handling the fail case in your corresponding code.